Testscripts/Windows/PERF-MEMCACHED-BENCHMARK.ps1

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the memcached License.
+param(
+    [String] $TestParams,
+    [object] $AllVMData,
+    [object] $CurrentTestData
+)
+
+function Main {
+    param (
+        $TestParams,
+        $AllVMData
+    )
+    # Create test result
+    $CurrentTestResult = Create-TestResultObject
+    $resultArr = @()
+
+    try {
+        $noClient = $true
+        $noServer = $true
+
+        # role-0 vm is considered as the server-vm
+        # role-1 vm is considered as the client-vm
+        foreach ($vmData in $allVMData) {
+            if ($vmData.RoleName -imatch "server" -or $vmData.RoleName -imatch "role-0") {
+                $serverVMData = $VmData
+                $noServer = $false
+            } elseif ($vmData.RoleName -imatch "client" -or $vmData.RoleName -imatch "role-1") {
+                $clientVMData = $VmData
+                $noClient = $false
+            }
+        }
+        if ($noClient -or $noServer) {
+            Throw "Client or Server VM not defined. Be sure that the SetupType has 2 VMs defined"
```

Comment:
Nah, wording!!!. This is not something the tester can customize. The VM type is defined in static XML. The possible scenario is VM provisioning failure.

Reply:
Fixed